compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -600,7 +605,8 @@ class Definitions {
                       paramNames = Nil, paramInfos = Nil))
                 }
             }
-            constr.info = newInfo
+            val argConstr = constr.copy().entered
```

Comment:
could you also edit the commit message so that it fixes #7174?

Reply:
yup!